Sources/Penny/Services/CacheService/DefaultCacheService.swift

```diff
@@ -0,0 +1,51 @@
+import Logging
+import AsyncHTTPClient
+import SotoS3
+
+actor DefaultCachesService: CachesService {
+
+    var cacheRepo: S3CacheRepository!
+    let logger = Logger(label: "DefaultCachesService")
+
+    private init() { }
+
+    static let shared = DefaultCachesService()
+
+    func initialize(httpClient: HTTPClient) {
+        let awsClient = AWSClient(httpClientProvider: .shared(httpClient))
+        self.cacheRepo = .init(awsClient: awsClient, logger: self.logger)
+    }
+
+    /// Get the storage from the repository and then delete it from the repository.
+    func getCachedInfoFromRepositoryAndPopulateServices() async {
+        do {
+            let storage = try await self.cacheRepo.get()
+            await storage.populateServices()
+            self.delete()
+        } catch {
+            logger.report("Couldn't get CacheStorage", error: error)
+        }
+    }
+
+    /// Delete the object from the repository.
+    /// We don't care if it succeeds or not.
+    private func delete() {
+        Task {
```

Comment:
Not saying I have an alternative suggestion, to be sure, but as a point of general being the pedantic creature I am, I should point out that @fabianfett would be steaming at this much mixing of structured and unstructured concurrency almost as much as you might expect me to do at, say, one of Tanner's infamous `Codable` misuses 😂